### ISSUE TITLE
Per-label classifier

### ISSUE THREAD

**Issue by MaxHalford** [online-ml/river#1408]
- It would be called `PerOutputClassifier`
- Much like https://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html#sklearn.multioutput.MultiOutputClassifier

**Comment by apagliero** [comment]
Hello @MaxHalford , very nice project. Congrats to you and all the contributors for the work done. 
@atroyanovsky and myself would like to contribute to the library and be assigned to this issue if possible.

**Comment by MaxHalford** [comment]
Hey @apagliero! Thanks for the enthusiasm. Knock yourself out!

**Comment by amadou1998** [comment]
How is this comming along? Currently interested in using a PerOutputClassifier

**Comment by amadou1998** [comment]
Should be relatively trivial I suggest: `class OneVsOnePerLabelClassifier(base.Wrapper, base.Classifier):

    def __init__(self, classifier):
        self.classifier = classifier
        new_clf = functools.partial(copy.deepcopy, classifier)
        self.classifiers = collections.defaultdict(new_clf)
        self.classes = set()

    @property
    def _wrapped_model(self):
        return self.classifier

    @property
    def _multiclass(self):
        return True

    @classmethod
    def _unit_test_params(cls):
        yield {"classifier": linear_model.LogisticRegression()}

    def learn_one(self, x, y, **kwargs):
        for label_name, y_label in y.items():
            self.classes.add(label_name)
            self.classifiers[label_name].learn_one(x, y_label)

    def predict_one(self, x, **kwargs):
        predictions = {}
        for label in self.classifiers:
            predictions[label] = self.classifiers[label].predict_one(x)

        return predictions
`

**Comment by apagliero** [comment]
Hi everyone. Sorry for the late update, unfortunately we've been too busy and haven't had time to progress on this issue.

**Comment by Siege-Cobby** [comment]
Hi @MaxHalford, I just merged #1409 (PerOutputRegressor), which is the regressor sibling of this one. The classifier version would follow the same pattern almost exactly, so I could pick it up and work it sometime this week. Do you mind if i get on with it?

**Comment by MaxHalford** [comment]
Yes please!
